Modix.Data/Models/Core/MessageEntity.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace Modix.Data.Models.Core
+{
+    public class MessageEntity
+    {
+        [Key]
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public long Id { get; set; }
+
+        [Required]
+        public ulong MessageId { get; set; }
+
+        [Required]
+        public ulong GuildId { get; set; }
+
+        [Required]
+        public ulong ChannelId { get; set; }
+
+        [Required]
+        public ulong UserId { get; set; }
```

Comment:
This should be `AuthorId`, for clarity.

Reply:
Also, you should have a navigation property `Author`

Reply:
I considered a relationship between this table and `GuildUsers`, but decided to remove it because it didn't look like it was safe to assume a guild user would exist before the insert, and because I didn't see a value add from it. Also, I'm not sure it's safe if IDs in other tables are stored as `bigint` instead of `numeric(20, 0)`.

Reply:
The point about foreign keys is almost certainly valid. I'll add a note to the issue about adding the relationship then.

The user SHOULD exist simply by the fact that we track users on a per-message basis already, but we can definitely avoid the risk of the user not being known. I'd say we simply have `UserTrackingBehavior` not listen to `MessageReceived` and have only `MessageLogBehavior` do that, and call `TrackUserAsync` for the message author, before logging the message.

The advantage to having the relationship is that we can pull back detailed information about a message all in one query, for use in UI's. I.E. if you wanna display the name of the author, you don't have to a separate call to get the name of the user from its id, which would either guarantee another DB hit if we pull from the DB, or risk the user's info not being found if we query from Discord.NET.

On a more fundamental note, the `GuildUsers` and `GuildChannels` tables are specifically intended to track info about all users and channels within any guild the bot serves. If there's a situation where the user or channel is NOT present there, I think that represents a separate bug.